### ISSUE TITLE
dose_main crashes when reading parameterized member in Safir.Dob.InjectionProperty

### ISSUE THREAD

**Issue by DonOregano** [SafirSDK/safir-sdk-core#291]
A customer is reporting an issue where dose_main crashes on startup in 6.1. After removing two property mappings everything works again.

Mapping 1:
```xml
<?xml version="1.0" encoding="utf-8" ?>
<propertyMapping xmlns="urn:safir-dots-unit"
                 xmlns:xsd="http://www.w3.org/2001/XMLSchema-instance">
   <property>Safir.Dob.InjectionProperty</property>
   <class>XXX.YYY.SomeNamespace.SomeEntity</class>
    <memberMapping>
        <member>
            <propertyMember>Injection</propertyMember>
            <valueRef>
                <name>XXX.YYY.Parameters.NodeDataPersistency</name>
            </valueRef>
        </member>
    </memberMapping>
</propertyMapping>
```

Mapping 2:
```xml
<?xml version="1.0" encoding="utf-8" ?>
<propertyMapping xmlns="urn:safir-dots-unit" xmlns:xsd="http://www.w3.org/2001/XMLSchema-instance">
    <property>Safir.Dob.InjectionProperty</property>
    <class>XXX.YYY.SomeNamespace.AnotherEntity</class>
    <memberMapping>
        <member>
            <propertyMember>Injection</propertyMember>
            <value>Injectable</value>
        </member>
    </memberMapping>
</propertyMapping>
```

The parameter referred to by the first mapping is
```xml
    <parameter>
      <summary>Parameter controls if node handles persistency of the data</summary>
      <name>NodeDataPersistency</name>
      <type>Safir.Dob.InjectionKind</type>
      <value>Injectable</value>
    </parameter>
```

**Comment by DonOregano** [comment]
```
[15:06:21.344777] dose_main entering main()
[15:06:31.905995] Opening dose shared memory
[15:06:32.186796] Loading DOTS shared memory
[15:06:32.202396] Reading DistributionScope properties to ascertain local types
[15:06:32.202396] Local type  - HERE a lot of entities
[15:06:32.202396] Local type  - Also ours 9 dous listed which has DistributedScope set to Local
.
.
[15:06:32.202396] Local type Safir.SituationPicture.CenterMapViewToObjectRequest
[15:06:32.217996] COM: -------------------------------------------------
[15:06:32.217996] COM: Communication initiated
[15:06:32.217996] COM:     id:    1572046111149935161
[15:06:32.217996] COM:     name:    MyNode
[15:06:32.217996] COM:     data address: 127.0.0.1:40000
[15:06:32.217996] COM:     control address:
[15:06:32.217996] COM:     multicast: 224.11.11.11:24000
[15:06:32.217996] COM:     using multicast: true
[15:06:32.217996] COM: -------------------------------------------------
[15:06:32.217996] Initializing dose_internal from dose_main
[15:06:32.264796] SP: AddSubscriber
[15:06:32.264796] SP: AddSubscriber calling Connect
[15:06:32.264796] Type Safir.Application.BackdoorCommand is ContextShared.
[15:06:32.389596] ALERT: dose_main has generated a dump to:
[15:06:32.389596] C:\ProgramData/safir-sdk-core/dump/\79f50a18-f200-402e-aefb-91e78b2b7e08.dmp
[15:06:32.389596] Please send this file to your nearest Dob developer, along with
[15:06:32.389596] relevant information about what version of Safir SDK you are using
```

Loading the dmp file points to DotsC_GetEnumerationParameter. Of course the dump is from release code, so the values of the local variables are unreliable.
A theory is that pd becomes NULL, and accessing that is why we get the error.
![screenshot_2016-03-23_15-09-27](https://cloud.githubusercontent.com/assets/3583175/13987744/467197f6-f109-11e5-8f9f-bd9661ca7338.png)

**Comment by DonOregano** [comment]
I've attempted to insert the mappings into a bare installation of Safir SDK Core, but everything works fine in that case, so there is something a bit more involved going on.

**Comment by DonOregano** [comment]
The customer is able to run dots_configuration_check without problems.

**Comment by DonOregano** [comment]
Further investigation reveals that valueRef elements inside dom files do not work at all. In fact, they may cause crashes or, even worse, use of random data.
A fix (1bc411e) is pending review.

**Comment by DonOregano** [comment]
The above fixes and tests solves this issue.
